src/main/java/org/folio/processing/mapping/mapper/util/AcceptedValuesUtil.java

```diff
@@ -0,0 +1,148 @@
+
+  private static final Map<String, Function<MappingParameters, List<?>>> ruleNameToMappingParameter = Map.ofEntries(
+    entry(HOLDINGS_PERMANENT_LOCATION_ID, MappingParameters::getLocations),
+    entry(HOLDINGS_TEMPORARY_LOCATION_ID, MappingParameters::getLocations),
+    entry(STATUS_ID, MappingParameters::getInstanceStatuses),
+    entry(NATURE_OF_CONTENT_TERM_ID, MappingParameters::getNatureOfContentTerms),
+    entry(INSTANCE_RELATIONSHIP_TYPE_ID, MappingParameters::getInstanceRelationshipTypes),
+    entry(HOLDINGS_TYPE_ID, MappingParameters::getHoldingsTypes),
+    entry(CALL_NUMBER_TYPE_ID, MappingParameters::getCallNumberTypes),
+    entry(ILL_POLICY_ID, MappingParameters::getIllPolicies),
+    entry(STATISTICAL_CODE_ID, AcceptedValuesUtil::getStatisticalCode),
+    entry(NOTE_TYPE, MappingParameters::getHoldingsNoteTypes),
+    entry(RELATIONSHIP_ID, MappingParameters::getElectronicAccessRelationships),
+    entry(MATERIAL_TYPE_ID, MappingParameters::getMaterialTypes),
+    entry(ITEM_CALL_NUMBER_TYPE_ID, MappingParameters::getCallNumberTypes),
+    entry(ITEM_DAMAGED_STATUS_ID, MappingParameters::getItemDamageStatuses),
+    entry(ITEM_NOTE_TYPE_ID, MappingParameters::getItemNoteTypes),
+    entry(PERMANENT_LOAN_TYPE_ID, MappingParameters::getLoanTypes),
+    entry(TEMPORARY_LOAN_TYPE_ID, MappingParameters::getLoanTypes),
+    entry(ITEM_PERMANENT_LOCATION_ID, MappingParameters::getLocations),
+    entry(ITEM_TEMPORARY_LOCATION_ID, MappingParameters::getLocations),
+    entry(CONTRIBUTOR_NAME_TYPE_ID, MappingParameters::getContributorNameTypes),
+    entry(ORDER_LOCATION, MappingParameters::getLocations),
+    entry(ORDER_MATERIAL_TYPE, MappingParameters::getMaterialTypes),
+    entry(VENDOR, MappingParameters::getOrganizations),
+    entry(MATERIAL_SUPPLIER, MappingParameters::getOrganizations),
+    entry(ACCESS_PROVIDER, MappingParameters::getOrganizations),
+    entry(DONOR_ORGANIZATION_IDS, AcceptedValuesUtil::getDonorOrganizationsFromMappingParameters),
+    entry(ACQUISITION_UNIT_IDS, MappingParameters::getAcquisitionsUnits),
+    entry(BILL_TO, MappingParameters::getTenantConfigurationAddresses),
+    entry(SHIP_TO, MappingParameters::getTenantConfigurationAddresses),
+    entry(PRODUCT_ID_TYPE, MappingParameters::getIdentifierTypes),
+    entry(ACQUISITION_METHOD, MappingParameters::getAcquisitionMethods),
+    entry(FUND_ID, MappingParameters::getFunds),
+    entry(EXPENSE_CLASS_ID, MappingParameters::getExpenseClasses));
+
+  private AcceptedValuesUtil() {}
+
+  public static Map<String, String> getAcceptedValues(String ruleName, MappingParameters mappingParameters) {
+    HashMap<String, String> acceptedValues = new HashMap<>();
+
+    if (ruleName == null || !ruleNameToMappingParameter.containsKey(ruleName)) {
+      return acceptedValues;
+    }
+
+    List<?> mappingParameter = ruleNameToMappingParameter.get(ruleName).apply(mappingParameters);
+
+    mappingParameter.forEach(o -> {
```

Comment:
@RomanChernetskyi  Please rename variable "o" -> "mappingParameter" (for example)

Reply:
Done thank you